compiler/src/test/java/com/squareup/anvil/compiler/dagger/ComponentDetectorCheckTest.kt

```diff
@@ -0,0 +1,114 @@
+package com.squareup.anvil.compiler.dagger
+
+import com.google.common.truth.Truth.assertThat
+import com.tschuchort.compiletesting.KotlinCompilation.ExitCode.COMPILATION_ERROR
+import com.tschuchort.compiletesting.KotlinCompilation.Result
+import org.junit.Test
+
+class ComponentDetectorCheckTest {
+
+  @Test fun `a Dagger component causes an error`() {
+    compile(
+        """
+        package com.squareup.test
+        
+        import dagger.Component
+        
+        @Component
+        interface ComponentInterface
+    """
```

Comment:
Nit: different start/end indentation.

Reply:
Will fix that everywhere in a follow-up PR.

Reply:
https://github.com/square/anvil/pull/69